avm/ptn/authorization/pim-role-assignment/main.bicep

```
@@ -0,0 +1,162 @@
+
+@sys.description('Optional. Ticket Info of the role eligibility.')
+param ticketInfo ticketInfoType?
+
+@sys.description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to.')
+param condition string = ''
+
+@sys.description('Optional. Version of the condition. Currently accepted value is "2.0".')
+@allowed([
+  '2.0'
+])
+param conditionVersion string = '2.0'
+
+@sys.description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.authorization-pimroleassignment.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+  location: location
+}
+
+module roleAssignment_mg 'modules/management-group.bicep' = if (empty(subscriptionId) && empty(resourceGroupName)) {
+  name: '${uniqueString(deployment().name, location)}-PimRoleAssignment-MG-Module'
+  scope: managementGroup(managementGroupId)
+  params: {
+    roleDefinitionIdOrName: roleDefinitionIdOrName
+    principalId: principalId
+    managementGroupId: managementGroupId
+    requestType: requestType
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    targetRoleAssignmentScheduleId: targetRoleAssignmentScheduleId
+    targetRoleAssignmentScheduleInstanceId: targetRoleAssignmentScheduleInstanceId
+    ticketInfo: ticketInfo
+    conditionVersion: conditionVersion
+    condition: !empty(condition) ? condition : ''
+    pimRoleAssignmentType: pimRoleAssignmentType
+  }
+}
+
+module roleAssignment_sub 'modules/subscription.bicep' = if (!empty(subscriptionId) && empty(resourceGroupName)) {
+  name: '${uniqueString(deployment().name, location)}-PimRoleAssignment-Sub-Module'
+  scope: subscription(subscriptionId)
+  params: {
+    roleDefinitionIdOrName: roleDefinitionIdOrName
+    principalId: principalId
+    subscriptionId: subscriptionId
+    requestType: requestType
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    ticketInfo: ticketInfo
+    conditionVersion: conditionVersion
+    condition: !empty(condition) ? condition : ''
```

```suggestion
    condition: condition
```
As you gave the parmaeter a default value of `''`, I guess this check is redundant 😄

---

Resolved